test/run-nools-lib.js

```diff
@@ -42,8 +47,8 @@ const runNoolsLib = ({ c, targets, tasks }) => {
     },
   };
 
-  targetEmitter(context.targets, context.c, context.Utils, context.Target, context.emit);
-  taskEmitter(context.tasks, context.c, context.Utils, context.Task, context.emit);
+  targetEmitter(context.targets, context.c, context.Utils(), context.Target, context.emit);
```

Comment:
```suggestion
  targetEmitter(context.targets, context.c, utilsMock || context.Utils, context.Target, context.emit);
```
I think we can keep the old Utils as object with a comment saying that it's deprecated and eventually will be removed, so other devs don't add more logic to that, and then make the conditional here.